src/pydata_sphinx_theme/utils.py

```diff
@@ -58,3 +60,79 @@ def maybe_warn(app: Sphinx, msg, *args, **kwargs):
     should_warn = theme_options.get("surface_warnings", False)
     if should_warn:
         SPHINX_LOGGER.warning(msg, *args, **kwargs)
+
+
+def set_secondary_sidebar_items(
+    app: Sphinx, pagename: str, templatename: str, context, doctree
+) -> None:
+    """Set the secondary sidebar items to render for the given pagename."""
+    if "theme_secondary_sidebar_items" in context:
+        templates = context["theme_secondary_sidebar_items"]
+        if isinstance(templates, dict):
+            templates = _get_matching_sidebar_items(pagename, templates)
+
+        context["secondary_sidebar_items"] = _update_and_remove_templates(
+            app,
+            context,
+            templates,
+            "theme_secondary_sidebar_items",
+        )
+
+
+def _update_and_remove_templates(
+    app: Sphinx, context, templates: Union[List, str], template_key: str
+) -> List[str]:
+    """Update templates to include html suffix if needed; remove templates which render empty."""
+    # Break apart `,` separated strings so we can use , in the defaults
+    if isinstance(templates, str):
+        templates = [ii.strip() for ii in templates.split(",")]
+    # Add `.html` to templates with no suffix
+    suffixed_templates = []
+    for template in templates:
+        if os.path.splitext(template)[1]:
+            suffixed_templates.append(template)
+        else:
+            suffixed_templates.append(f"{template}.html")
+
+    ctx = copy.copy(context)
+    ctx.update({template_key: suffixed_templates})
+
+    # Check whether the template renders to an empty string; remove if this is the case
+    filtered_templates = []
+    for template in suffixed_templates:
+        rendered = app.builder.templates.render(template, ctx)
+        if len(rendered.strip()) != 0:
+            filtered_templates.append(template)
+
+    return filtered_templates
+
+
+def _get_matching_sidebar_items(
+    pagename: str, sidebars: Dict[str, List[str]]
+) -> List[str]:
+    """Get the matching sidebar templates to render for the given pagename.
+
+    This function was adapted from sphinx.builders.html.StandaloneHTMLBuilder.add_sidebars.
+
+    Cached to avoid overhead when html context event is fired multiple times on the same page.
```

Comment:
In what way is it cached?  I'm only really familiar with `@lru_cache` decorators, so if you're doing it another way I probably just don't recognize it.

Reply:
Honestly, I don't remember why I wrote this. Removed.